### ISSUE TITLE
Edit field in quick picker is inaccessible to screen readers

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#101183]
CC: @isidorn

Issue Type: <b>Bug</b>

1. With NVDA running, press ctrl+p or ctrl+shift+p. Then type some text.

Expected: pressing backspace or using the arrow keys reads the relevant characters as with usual edit fields when focus is on the edit field which controls the options that appear.
Actual: nothing is heard

VS Code version: Code - Insiders 1.47.0-insider (0913b1aa43191d8af0ccb4a133d9a8d7c1a81d69, 2020-06-23T09:03:35.861Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (4.43GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|17%|
</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
notes|can|1.0.1
vscode-markdownlint|Dav|0.36.1
vscode-eslint|dba|2.1.5
gitlens|eam|10.2.2
prettier-vscode|esb|5.1.0
Go|gol|0.14.4
rest-client|hum|0.24.1
python|ms-|2020.6.90262
remote-wsl|ms-|0.44.4
cpptools|ms-|0.28.3
vsnotes|pat|0.7.1
rust|rus|0.7.8
markdown-preview-enhanced|shd|0.5.12
vscode-webhint|web|1.5.6

</details>

**Comment by isidorn** [comment]
I tried this out and can reproduce with NVDA.
However it works just fine with Orca and VoiceOver. So I think there seems to be an issue on the NVDA side here.
@leonardder @feerrenrut Is it possible to look into this from the NVDA side. Repro should be quite easy with the comments from above. 
We are setting `activeDescendent` to set focus on the child element so that might be confusing NVDA?

**Comment by feerrenrut** [comment]
Is there a related NVDA issue? I couldn't find one. It would be really helpful to have a test sample to ensure we look at the right thing.

**Comment by isidorn** [comment]
@feerrenrut Sorry for the slow response, I was on vacation last week. I have created this issue https://github.com/nvaccess/nvda/issues/13037 let's continue the discussion there. Thanks!

**Comment by leonardder** [comment]
I can reproduce this with Narrator as well. I'd be curious about JAWS. As per https://github.com/nvaccess/nvda/issues/13037#issuecomment-962016218 I'm not sure whether saying this is NVDA specific is a valid conclusion, may be it is Windows specific though.

**Comment by leonardder** [comment]
It might be good too look ath the [Aria best practice](https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html) for development guidelines. Those seem to behave correctly on Windows.

**Comment by Neurrone** [comment]
I can confirm this problem still exists.

**Comment by isidorn** [comment]
I can also reproduce. I also tried with the Exploration build that uses the latest Chrome and I can still reproduce.
@TylerLeonhardt can you check if we for example do not set the `active-descendent` does this no longer repro? Just so we make sure that is the thing that confuses screen readers on Windows. Maybe we need to introduce a workaround: when the user navigates left / right in the quick pick do not set the active descendant. 

@leonardder I still believe that the issue is on the Windows / Chrome / Screen reader side. And that NVDA should get events from Chrome if there is a cursor change on a parent of the active descendent.

**Comment by TylerLeonhardt** [comment]
This should be working now since for accessibility mode we do not set `aria-activedescendent` when the input box changes. https://github.com/microsoft/vscode/pull/176080

**Comment by TylerLeonhardt** [comment]
reopening as I am depending on aria-activedescendant again in https://github.com/microsoft/vscode/pull/179217

**Comment by TylerLeonhardt** [comment]
@rperez030 mentioned we could hook up `Shift` + `Tab` to remove the `aria-activedescendant` by maybe focusing none.

**Comment by rperez030** [comment]
@TylerLeonhardt  this is looking great to me. Some ideas for possible refinements.

focus management.

If I set focus to an element in the list, press shift +Tab and tab back to the list again, focus gets reset. In fact, it seems like the parent list receives focus at this point. There are also instances in which I need to Shift Tab twice to move focus from the list to the combobox, particularly if it is the second or third time I do it. Ideally focus would return to the item that was selected before pressing Shift +Tab. If this is not possible, I would then suggest that Tabs behaves like Down arrow key which focuses on the first element in the list.

controls accessible names and descriptions

Right now, the combobox has the generic name of 'input'. NVDA calculates the accessible description based on the placeholder text but JAWS doesn't. My suggestion: We could use a more descriptive name for the combobox, maybe "filter", and use aria-description for the placeholder text.
The list currently receives the accessible name from the placeholder text as we discussed, but that feels more like a description to me. suggestion: We could give the list a generic name (E.g. 'actions'), and use aria-description to communicate the placeholder text. Ideally that list will be named based on context depending on how the widget is used, but that may complicate things and open a door for future issues if it is not defined correctly.

Like I said, even without refinements, in my opinion, this feels even better than the initial implementation. thank you very much to everyone involved.

**Comment by TylerLeonhardt** [comment]
@rperez030 thanks for your feedback! Can you open new issues for each of these improvements so that I can tackle them independently?

I'm glad it's better for you. I may have to focus on other things I own at the moment, but I want to make sure these additional improvments are captured.

**Comment by rperez030** [comment]
Definitely. I will open separate issues. Maybe I can even tackle some of this myself with a bit of mentorship.
